src/viewer.js

```diff
@@ -1968,7 +1972,12 @@ function openTileSource( viewer, source ) {
         _this.viewport.resetContentSize( _this.source.dimensions );
     }
 
-    _this.source.overlays = _this.source.overlays || [];
+    if( _this.preserveOverlays ){
+        _this.source.overlays = _this.currentOverlays;
```

Comment:
If we do ``` _this.overlays ``` we have to remove the ```else``` and always initialize ```_this.source.overlays```, otherwise the second loop in ```loadOverlays()``` will complain, i've tested this and it works as well (even tho i don't fully understand the logic behind the ```[i + j] in loadOverlays()```